crates/biome_glob/src/editorconfig.rs

```diff
@@ -0,0 +1,702 @@
+//! `biome_glob::editorconfig` provides globbing functionality that follows:
+//!
+//! - the [specification](https://editorconfig.org/#wildcards), and
+//! - the [test suite](https://github.com/editorconfig/editorconfig-core-test/tree/master/glob).
+//!
+//! The current implementation has some limitations:
+//!
+//! - Nested alternates (choices) are not supported.
+//!   For example, `{a,b{a,b},d}` is not supported
+//! - Empty choice such as `{a,b,}` is not supported
+//! - `a**b` is translated to `a*/**/b`.
+//!   Thus, this  doesn't match `ab` or `axb`.
+//!
+//! ## Matching a path against a glob
+//!
+//! You can create a glob from a string using the `parse` method.
+//! Use [Glob::is_match] to match against anything that can be turned into a [std::path::Path], such as a string.
+//!
+//! In the following example we parse the string `"*.rs"` into a glob, and we match against two strings.
+//! `lib.rs` matches the glob because the path has a single path segment that ends with `.rs`.
+//! Conversely, `src/lib.rs` doesn't match because it has two path segments (`src` and `lib.rs`).
+//!
+//! ```
+//! use biome_glob::editorconfig::EditorconfigGlob;
+//!
+//! let glob: EditorconfigGlob = "*.rs".parse().expect("correct glob");
+//! assert!(glob.is_match("lib.rs"));
+//! assert!(!glob.is_match("src/lib.rs"));
+//! ```
+//!
+//! ## Matching against multiple globs
+//!
+//! When a path is expected to be matched against several globs,
+//! you should compile the path into a [crate::CandidatePath] using [crate::CandidatePath::new].
+//! [crate::CandidatePath] may speed up matching against several globs.
+//! To get advantage of the speed-up, you have to use the [CandidatePath::matches] method instead of [Glob::is_match].
```

Comment:
The example doesn't show how to use `matches`. Was that a mistake?

Reply:
Thanks!